RHDHPAI-808: Developer Lightspeed content RHDH (#1300)
redhat-developer/red-hat-developers-documentation-rhdh: modules/developer-lightspeed/proc-customizing-the-chat-history-storage.adoc

// ARCHIVED

:_mod-docs-content-type: PROCEDURE
[id="proc-customizing-the-chat-history-storage_{context}"]
= Customizing the chat history storage in {ls-short}

> By default, the {rcs-short} service stores chat history using an in memory database. This means that if you restart the Pod containing the server, the chat history is lost. You can manually configure {ls-short} to store the chat history persistently as a long-term backup with PostgreSQL by any of the following methods:

Reviewer: ```suggestion
By default, the {rcs-short} service stores chat history using an in-memory database. As a result, if you restart the Pod containing the server, the chat history is lost. You can manually configure {ls-short} to store the chat history persistently as a long-term backup with PostgreSQL by any of the following methods:
```

====

.Procedure
* When you are using {ls-short} on an Operator-installed {product-very-short} instance, in your {product-very-short} instance ConfigMap, update the `conversation-cache` field as shown in the following example:
+
[source,yaml]
----
 conversation_cache:
    type: postgres
    postgres:
      host: _<your_database_host>_
      port: _<your_database_port>_
      dbname: _<your_database_name>_
   	  user: _<your_user_name>_
	    password_path: postgres_password.txt
	    ca_cert_path: postgres_cert.crt
      ssl_mode: "require" 
----

* When you are using {ls-short} on a Helm-installed {product-very-short} instance, in your {product-very-short} instance `values.yaml` file, update the `conversation-cache` field as shown in the following example:
+
[source,yaml]
----
 conversation_cache:
    type: postgres
    postgres:
      host: _<your_database_host>_
      port: _<your_database_port>_
      dbname: _<your_database_name>_
   	  user: _<your_user_name>_
	    password_path: postgres_password.txt

> 	    ca_cert_path: postgres

Reviewer 2: ```suggestion
	    ca_cert_path: postgres_cert.crt
      ssl_mode: "require" 
```
